src/Sulu/Bundle/SecurityBundle/Entity/Role.php

```diff
@@ -110,6 +116,10 @@ public function getRole()
 
     public function getIdentifier()
     {
+        if ($this->anonymous) {
+            return AuthenticatedVoter::IS_ANONYMOUS;
```

Comment:
How should we handle this the identifier of our anonymous roles:

*Option A*:

Override the Identifier function and return the Symfony `IS_ANONYMOUS` role if anonymous flag is set.

*Option B*:

Change the unique constraint from [Role](https://github.com/sulu/sulu/blob/89518be214a030c68d12c16a459f2ef74beced4d/src/Sulu/Bundle/SecurityBundle/Resources/config/doctrine/Role.orm.xml#L12)  from `key` to `key,system`. So a role is unique inside its system.

*Option C*

Create a Role with a Identifier `ROLE_<SYSTEM>_ANONYMOUS`. This sadly lose the connection to the Symfony `IS_ANONYMOUS` flag an its `is_granted('IS_ANONYMOUS')` function, but not sure if we need that.

Reply:
If I think about it if we go with **Option A** or **Option B** the `getRoles` function should only return Roles for the current system. So the UserProvider would need to set a in memory flag on the user with `currentSystem`.

Reply:
Currently option A is implemented, right? Seems like the cleanest solution to me, but wouldn't we still have to make only the combination of `system` and `key` unique?

Reply:
> Currently option A is implemented, right?

Yes implemented is currently *Option A*.

> Seems like the cleanest solution to me, but wouldn't we still have to make only the combination of system and key unique?

No as the key is null in the database we don't need todo that for anonymous roles.

Reply:
If this works like that, I don't see any reason to change it :shrug: Would probably keep it as it is, and only change if we encounter any problems.

Reply:
Okay then I will got with that solution.